src/rules/javascript/complex_prefs_defaults.js

```diff
@@ -0,0 +1,26 @@
+import { COMPLEX_PREFS_DEFAULTS } from 'messages/javascript';
+
+export default function(context) {
+  var filename = context.getFilename();
+
+  if (filename.indexOf('defaults/preferences/') === 0) {
```

Comment:
Explaining why we do this inside the rule would be handy. I don't mind it being used outside the event function `CallExpression` but just explaining what purpose it serves would be nice.

Also: we should have a test to see what happens if no filename is supplied. We don't usually fail to pass a filename but in theory ESLint can take stuff from `stdin` and I just like covering bases. A simple check to see if `filename` is truthy in here along with the `indexOf` makes sense to me.